src/fiber/fiber.mli

```diff
@@ -381,6 +381,6 @@ with type 'a fiber := 'a t
 type fill = Fill : 'a Ivar.t * 'a -> fill
 
 (** [run t ~iter] runs a fiber until it terminates. [iter] is used to implement
-    the scheduler, it should block waiting for an event and return an ivar to
-    fill. *)
-val run : 'a t -> iter:(unit -> fill) -> 'a
+    the scheduler, it should block waiting for an event and return at least one
+    ivar to fill. *)
+val run : 'a t -> iter:(unit -> fill Nonempty_list.t) -> 'a
```

Comment:
I wonder if it makes sense to keep the singleton version as well: in most cases it's what's needed and it avoids unnecessary allocations of singleton lists.

Reply:
(Then we won't need any changes in `scheduler.ml`.)